internal/core/workers/operation_execution_worker/operation_execution_worker.go

```diff
@@ -76,125 +79,158 @@ func (w *OperationExecutionWorker) Start(ctx context.Context) error {
 			if errors.Is(err, context.Canceled) {
 				return nil
 			}
-
-			w.Stop(ctx)
 			reportOperationExecutionWorkerFailed(w.scheduler.Game, w.scheduler.Name, LabelNextOperationFailed)
 			return fmt.Errorf("failed to get next operation: %w", err)
 		}
 
-		loopLogger := w.logger.With(
-			zap.String(logs.LogFieldOperationID, op.ID),
-			zap.String(logs.LogFieldOperationDefinition, def.Name()),
-		)
-
-		if op.Status != operation.StatusPending {
-			loopLogger.Warn("operation is at an invalid status to proceed")
-
-			continue
-		}
-
-		executor, hasExecutor := w.executorsByName[def.Name()]
-		if !hasExecutor {
-			loopLogger.Warn("operation definition has no executor")
-
-			w.evictOperation(ctx, loopLogger, op)
-			reportOperationEvicted(w.scheduler.Game, w.scheduler.Name, op.DefinitionName, LabelNoOperationExecutorFound)
-
-			continue
+		err = w.executeOperation(op, def)
+		if err != nil {
+			switch {
+			case errors.Is(err, workererrors.ErrStartOperationFailed) ||
+				errors.Is(err, workererrors.ErrGrantLeaseFailed) ||
+				errors.Is(err, workererrors.ErrOperationWithInvalidStatus):
+				return err
+			}
 		}
+	}
+}
 
-		if !def.ShouldExecute(ctx, []*operation.Operation{}) {
-			w.evictOperation(ctx, loopLogger, op)
-			reportOperationEvicted(w.scheduler.Game, w.scheduler.Name, op.DefinitionName, LabelShouldNotExecute)
-			continue
-		}
+func (w *OperationExecutionWorker) executeOperation(op *operation.Operation, def operations.Definition) error {
+	loopLogger := w.logger.With(
+		zap.String(logs.LogFieldOperationID, op.ID),
+		zap.String(logs.LogFieldOperationDefinition, def.Name()),
+	)
 
-		loopLogger.Info("Starting operation")
+	if op.Status != operation.StatusPending {
+		loopLogger.Warn("operation is at an invalid status to proceed")
 
-		w.operationManager.AppendOperationEventToExecutionHistory(ctx, op, "Starting operation")
+		return workererrors.NewErrOperationWithInvalidStatus("operation is at and invalid status to proceed")
+	}
 
-		operationContext, operationCancellationFunction := context.WithCancel(ctx)
-		err = w.operationManager.GrantLease(operationContext, op)
-		if err != nil {
-			w.Stop(ctx)
-			reportOperationExecutionWorkerFailed(w.scheduler.Game, w.scheduler.Name, LabelStartOperationFailed)
-			operationCancellationFunction()
+	executor, hasExecutor := w.executorsByName[def.Name()]
+	if w.shouldEvictOperation(op, def, hasExecutor, loopLogger) {
+		return nil
+	}
 
-			op.Status = operation.StatusError
-			w.operationManager.AppendOperationEventToExecutionHistory(ctx, op, fmt.Sprintf("Failed to grant lease to operation, reason: %s", err.Error()))
+	err, executionErr := w.startAndExecuteOperationWithLease(op, def, loopLogger, executor)
```

Comment:
What do you think to unify these errors in only one?

Reply:
err and executionErr are different errors, not sure how we would unify them